test/models/user_test.rb

```diff
@@ -37,4 +37,235 @@ class UserTest < ActiveSupport::TestCase
     assert_not current_user.authorized_to?('app:resource1:read')
     assert_not current_user.authorized_to?('app:resource1:*')
   end
+
+  test 'can request users access to inventory groups' do
+    account = FactoryBot.create(:account)
+    current_user = FactoryBot.create(:user, account: account)
+    fake_rbac_api_response = RBACApiClient::AccessPagination.new(
+      data: [
+        RBACApiClient::Access.new(
+          permission: Rbac::INVENTORY_GROUPS_READ,
+          resource_definitions: [
+            {
+              attributeFilter: {
+                key: 'groups.id',
+                value: %w[
+                  78e3dc30-cec3-4b49-be2d-37482c74a9ac
+                  79e3dc30-cec3-4b49-be2d-37482c74a9ad
+                ],
+                operation: 'in'
+              }
+            }
+          ]
+        ),
+        RBACApiClient::Access.new(
+          permission: Rbac::INVENTORY_GROUPS_READ,
+          resource_definitions: [
+            {
+              attributeFilter: {
+                key: 'group.id',
+                value: '80e3dc30-cec3-4b49-be2d-37482c74a9ad',
+                operation: 'equal' # 'equal' is a supported operation
+              }
+            }
+          ]
+        ),
+        RBACApiClient::Access.new(
+          permission: Rbac::INVENTORY_GROUPS_READ,
+          resource_definitions: [
+            {
+              attributeFilter: {
+                key: 'groups.id',
+                value: [
+                  nil # ungrouped hosts
+                ],
```

Comment:
What if you have more entities here than the `nil`?

Reply:
Created a test for this.